lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -0,0 +1,241 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class Shift4Gateway < Gateway
+      self.test_url = 'https://utgapi.shift4test.com/api/rest/v1/'
+      self.live_url = 'https://utg.shift4api.net/api/rest/v1/'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://utg.shift4api.net'
+      self.display_name = 'Shift4'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :client_guid, :auth_token)
+        super
+      end
+
+      def access_token(options = {})
```

Comment:
Is this an access token that only ever needs to be generated once per client? If so, I think it would make sense to handle it as a credential. 

If it needs to be refreshed, that should take place as part of the `commit` method, rather than requiring users to call this method separately and then pass the result in to other transaction calls.